tool/structgen/src/main/java/org/openscience/cdk/structgen/maygen/Maygen.java

```diff
@@ -693,23 +692,51 @@ public void setSymbols(List<String> symbolList) {
         }
     }
 
+    /** Replace each sub-string in from[] with it's normalisation in to[]. Since
+     *  this is only used for formula validating and normalizing a parser would
+     *  be better. */
+    private static String replaceEach(String str, String[] from, String[] to) {
+        if (from.length != to.length)
```

Comment:
It will be better to use other implemrntation with better error handling.

Reply:
It would be better to not use this function at all. Using string munging to validate a formula is not nice. Adding in a third party library for a simple function - no